src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebApplicationTests.cs

```diff
@@ -313,6 +313,30 @@ public void WebApplicationBuilder_CanClearDefaultLoggers()
                 args.Payload.OfType<string>().Any(p => p.Contains(guid)));
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void WebApplicationBuilder_CanSetWebRootPaths(bool useSetter)
+        {
+            var builder = WebApplication.CreateBuilder();
+            var webRootPath = "www";
+            var fullWebRootPath = Path.Combine(Directory.GetCurrentDirectory(), webRootPath);
+
+            if (useSetter)
+            {
+                builder.Environment.WebRootPath = webRootPath;
+            }
+            else
+            {
+                builder.WebHost.UseWebRoot(webRootPath);
+                Assert.Equal(webRootPath, builder.WebHost.GetSetting("webroot"));
+            }
+
+            
+            var app = builder.Build();
+            Assert.Equal(fullWebRootPath, app.Environment.WebRootPath);
```

Comment:
Can we add a WebApplicationFunctionalTests that checks we have a working PhysicalFileProvider too?

Reply:
This is validated in `WebHostEnvironmentTests.SettingPathsSetsContentProviders` now. Not a full end-to-end test but validates that the providers are updated when the path is.